Improve codebase docs (#122)
holochain/android-service-runtime: CONTRIBUTING.md

2. Create and push a git tag with a matching format to trigger the release CI.

    The git tag format is: `service-vX.Y.Z`, where "X.Y.Z" is replaced with the version number.

## CI

> Note that CI checks for building `tauri-plugin-holochain-service` and `tauri-plugin-holochain-service-client` do *not* publish the client & service libraries to a local maven repository, but instead use the versions published to Maven Central.

Reviewer: ```suggestion
Note that the CI jobs for building `tauri-plugin-holochain-service` and `tauri-plugin-holochain-service-client` do *not* publish the client & service libraries to a local maven repository, but instead use the versions published to Maven Central.
```

Author: Tried to make it a bit clearer -- the main point is the sentence after: https://github.com/holochain/android-service-runtime/pull/122/commits/11771f7d42080ac53b8aaa2ce3eeab430dce56e4

Reviewer: :+1:

> If your PR makes changes to the client or service library, it will need to be published to Maven Central, before the CI checks of `tauri-plugin-holochain-service` and `tauri-plugin-holochain-service-client` will pass.

Author: ```suggestion
If your PR makes changes to the client or service library, it will need to be published to Maven Central, before the CI jobs `build-tauri-plugins` will pass.
```
